lib/ansible/executor/playbook_executor.py

```diff
@@ -134,7 +134,7 @@ def run(self):
                         # we are just doing a listing
                         entry['plays'].append(new_play)
 
-                    else:
+                    elif play.evaluate_tags(self._options.tags, self._options.skip_tags, all_vars):
```

Comment:
this does not take into account tags at the task level, just because a play has a tag and tasks will inherit it it does not mean that is deterministic of a task needing to execute or not (same with includes/roles).